WcaOnRails/app/webpacker/components/CompetitionOverview/CompetitionTable.js

```diff
@@ -0,0 +1,190 @@
+        position="top center"
+        size="tiny"
+      />
+    );
+  }
+
+  return (
+    <Popup
+      trigger={<Icon className="user plus green" />}
+      content={I18n.t('competitions.index.tooltips.registration.open')}
+      position="top center"
+      size="tiny"
+    />
+  );
+}
+
+function DateIcon({ comp, showRegistrationStatus, sortByAnnouncement }) {
+  let tooltipInfo = '';
+  let iconClass = '';
+
+  if (comp.isProbablyOver) {
+    if (comp.resultsPosted) {
+      tooltipInfo = I18n.t('competitions.index.tooltips.hourglass.posted');
+      iconClass = 'check circle result-posted-indicator';
+    } else {
+      tooltipInfo = I18n.t('competitions.index.tooltips.hourglass.ended', { days: I18n.t('common.days', { count: calculateDayDifference(comp.start_date, comp.end_date, 'past') }) });
+      iconClass = 'hourglass end';
+    }
+  } else if (comp.inProgress) {
+    tooltipInfo = I18n.t('competitions.index.tooltips.hourglass.in_progress');
+    iconClass = 'hourglass half';
+  } else if (sortByAnnouncement) {
+    tooltipInfo = I18n.t('competitions.index.tooltips.hourglass.announced_on', { announcement_date: comp.announcedDate });
+    iconClass = 'hourglass start';
+  } else if (showRegistrationStatus) {
+    return <RegistrationStatus comp={comp} />;
+  } else {
+    tooltipInfo = I18n.t('competitions.index.tooltips.hourglass.starts_in', { days: I18n.t('common.days', { count: calculateDayDifference(comp.start_date, comp.end_date, 'future') }) });
+    iconClass = 'hourglass start';
+  }
+
+  return (
+    <Popup
+      trigger={<Icon className={iconClass} />}
+      content={tooltipInfo}
+      position="top center"
+      size="tiny"
+    />
+  );
+}
+
+// Currently, the venue attribute of a competition object can be written as markdown,
+// and using third party libraries like react-markdown to parse it requires to much work
+function VenueMarkdown({ venueText }) {
+  const openBracketIndex = venueText.indexOf('[');
+  const closeBracketIndex = venueText.indexOf(']', openBracketIndex);
+  const openParenIndex = venueText.indexOf('(', closeBracketIndex);
+  const closeParenIndex = venueText.indexOf(')', openParenIndex);
+
+  if (openBracketIndex === -1 || closeBracketIndex === -1
+    || openParenIndex === -1 || closeParenIndex === -1) {
+    return <p>{venueText}</p>;
+  }
+
+  return (
+    <a href={venueText.slice(openParenIndex + 1, closeParenIndex)} target="_blank" rel="noreferrer">
+      <p>{venueText.slice(openBracketIndex + 1, closeBracketIndex)}</p>
+    </a>
+  );
+}
+
+function LoadedDisplay({ numCompetitions }) {
+  return (
+    <List.Item style={{ textAlign: 'center' }}>
+      {numCompetitions > 0 ? I18n.t('competitions.index.no_more_comps') : I18n.t('competitions.index.no_comp_found')}
+    </List.Item>
+  );
+}
+
+function CompetitionTable({
+  competitionData,
+  title,
+  showRegistrationStatus,
+  showCancelled,
+  selectedEvents,
+  loading,
+  loaded,
```

Comment:
I haven't looked at where this component is being called yet, but components should make sense in isolation and I'm a bit confused why there's `loading` and `loaded` - and why the main data `competitionData` seems to be presented regardless of the `load...` values.